apps/staking/app/register/[nodeId]/NodeRegistrationForm.tsx

```diff
@@ -0,0 +1,516 @@
+    };
+  }, [setIsBalanceVisible]);
+
+  return (
+    <Form {...form}>
+      <form onSubmit={form.handleSubmit(onSubmit)} className="flex flex-col gap-4">
+        {!isRemoteFlagLoading && isRegistrationPausedFlagEnabled ? (
+          <span>Registrations are disabled</span>
+        ) : null}
+        {stakedNode ? (
+          <>
+            <span className="mb-4 text-lg font-medium">
+              {dictionary.rich('notFound.foundRunningNode')}
+            </span>
+            <StakedNodeCard node={stakedNode} networkTime={networkTime} blockHeight={blockHeight} />
+          </>
+        ) : runningNode ? (
+          <>
+            <span className="mb-4 text-lg font-medium">
+              {dictionary('notFound.foundRunningNodeOtherOperator')}
+            </span>
+            <StakedNodeCard
+              node={runningNode}
+              networkTime={networkTime}
+              blockHeight={blockHeight}
+              hideButton
+            />
+          </>
+        ) : openNode ? (
+          <>
+            <span className="mb-4 text-lg font-medium">{dictionary('notFound.foundOpenNode')}</span>
+            <OpenNodeCard node={openNode} forceSmall />
+          </>
+        ) : null}
+        <ActionModuleRow
+          label={sessionNodeDictionary('publicKeyShort')}
+          tooltip={sessionNodeDictionary('publicKeyDescription')}
+        >
+          <PubKey pubKey={node.pubkey_ed25519} force="collapse" alwaysShowCopyButton />
+        </ActionModuleRow>
+        <ActionModuleRow
+          label={registerCardDictionary('type')}
+          tooltip={registerCardDictionary('typeDescription')}
+        >
+          <div className="flex flex-row items-center gap-1 align-middle">
+            {registerCardDictionary(isSolo ? 'solo' : 'multi')}
+            <ActionModuleTooltip>
+              {registerCardDictionary(isSolo ? 'soloDescription' : 'multiDescription')}
+            </ActionModuleTooltip>
+          </div>
+        </ActionModuleRow>
+        <ActionModuleRow
+          label={dictionary('preparedAtTimestamp')}
+          tooltip={dictionary('preparedAtTimestampDescription')}
+        >
+          <Tooltip
+            tooltipContent={formatDate(preparationDate, {
+              dateStyle: 'full',
+              timeStyle: 'full',
+            })}
+          >
+            <div className="cursor-pointer">
+              {formatLocalizedRelativeTimeToNowClient(preparationDate, { addSuffix: true })}
+            </div>
+          </Tooltip>
+        </ActionModuleRow>
+        <ActionModuleRow
+          label={sessionNodeDictionary('operatorFee')}
+          tooltip={sessionNodeDictionary('operatorFeeDescription')}
+        >
+          <FormField
+            control={form.control}
+            name="operatorFee"
+            render={({ field: { value, onChange, ...field } }) => {
+              const thousandsSeparator = decimalDelimiter === '.' ? ',' : '.';
+              const formatInputText = (value: string) => {
```

Comment:
Could this function be extracted out of the component?

Reply:
yes, but its cursed and not worth the time, maybe in the next code cleanup

Reply:
for more clarity, extracting the form item render function is hell